test/test_address_book.py

```diff
@@ -120,6 +121,30 @@ def test_unparsable_files_can_be_skipped(self):
             'WARNING:khard.address_book:1 of 1 vCard files of address book '
             'test could not be parsed.')
 
+    def test_env_var_paths(self):
+        var_name = "KHARD_FOO"
+
+        # Unset env vars shouldn't expand.
+        if var_name in os.environ:
+            os.environ.pop(var_name)
+        try:
+            address_book.VdirAddressBook(
+                "test", "test/fixture/test.abook${}".format(var_name))
+            self.assertTrue(False)
+        except FileNotFoundError as err:
+            self.assertIn("test/fixture/test.abook${}".format(var_name),
+                          err.args[0])
+
+        # Env vars set to empty string should expand to empty string.
+        os.environ[var_name] = ""
+        address_book.VdirAddressBook(
+            "test", "test/fixture/test.abook${}".format(var_name))
+
+        # Env vars set to nonempty string should expand appropriately.
+        os.environ[var_name] = "test/fixture"
+        address_book.VdirAddressBook(
```

Comment:
Again, extra method and asset missing.

Reply:
I'm not sure what you mean by "asse[r]t missing".  I instead added an assert to make sure that the abook got created properly.

Reply:
I meant that you did execute some code from khard but did not assert anything
afterwards.

In my opinion the generic description of how a test case looks like is
```python
def test_whatever(self):
    # prepare some objects that we might need
    x, y, z = 1, 2, 3
    # set up some mocks if we need them
    with unittest.mock.patch(...):
	# execute the code we want to test
        result = the_function_to_test(x, y, z)
    # check that the correct thing happend
    self.assertEqual(result, "what I expected")
```

And a unit test should only check *one* thing. So it is better to make small
test functions. But you did already split the test function so I think you
understood.